scalding-core/src/main/scala/com/twitter/scalding/typed/WithReducers.scala

```diff
@@ -24,6 +24,14 @@ trait HasReducers {
 }
 
 /**
+ * used for types that must know how many reducers they need
+ * e.g. Sketched
+ */
+trait MustHaveReducers[+This <: MustHaveReducers[This]] extends HasReducers {
```

Comment:
why do we need the F-bounded polymorphism here? The trait never references `This`. I think this complicates things a bit. Can it be removed?